.github/workflows/pre-merge-ci.yml

```diff
@@ -0,0 +1,10 @@
+name: Pre-Merge Integration tests
+
+on: 
+  pull_request:
+    types: [ labeled, opened, synchronize, reopened ]
```

Comment:
If a PR is labelled, will every push to the PR trigger this? And will I see the action being run on the PR page, or only in the actions tab?

Reply:
Yes, every push to a PR that is labeled will trigger a run of the integration test. If you know you are going to push multiple commits and don't want to run the tests, you can remove the label, push your commits, then add the label back to pass the tests before merging.